src/__tests__/wait-for.js

```diff
@@ -34,6 +34,17 @@ test('uses generic error if there was no last error', async () => {
   expect(result).toMatchInlineSnapshot(`[Error: Timed out in waitFor.]`)
 })
 
+test('uses full stack error trace when showOriginalStackTrace present', async () => {
+  const error = new Error('Throws the full stack trace')
+  const result = await waitFor(
+    () => {
+      throw error
+    },
+    {timeout: 8, interval: 5, showOriginalStackTrace: true},
+  ).catch(e => e)
+  expect(result).toBe(error)
```

Comment:
It's possible that the `error.stack` could've been manipulated (if we decide to change the implementation in the future). So I'd change this:

```suggestion
  expect(result.stack).toBe(error.stack)
```